Include pa log in test_status assertion failure messages.

ebd2 opened the pull request:
test_status relies on comparing the actual output of presto-admin server status
to the expected output. Unfortunately, the status command is written in such a
way that some failures in PrestoClient are logged and otherwise silently
ignored. This results in incomplete output and an assertion failure with no
information about what the failure actually was. Since this is intermittent
and usually happens on Jenkins jobs, we don't get a chance to look at the logs
manually. The solution is to include them in the assertion failure message.

petroav commented on tests/base_test_case.py:
```diff
@@ -115,12 +115,16 @@ def assertRaisesRegexp(self, expected_exception, expected_regexp,
+        try:
+            assert_function(actual, expected)
+        except AssertionError:
+            assert_function(actual, expected, msg=msg_func())
```
I'm confused as to why you need to call `assert_function()` here. Can't you just call `msg_func()`?

ebd2 replied:
We get a slightly more useful traceback than if we just call self.fail:

```
Failure
Traceback (most recent call last):
  File "/home/eric/presto-admin-clean/tests/unit/test_base_test_case.py", line 39, in testBar
    self.assertEquals(1, 2, msg='Hello, world!')
AssertionError: Hello, world!


Failure
Traceback (most recent call last):
  File "/home/eric/presto-admin-clean/tests/unit/test_base_test_case.py", line 42, in testBaz
    self.fail(msg='Hello, world!')
AssertionError: Hello, world!
```

ebd2 replied:
Actually, that isn't really true, since the actual assert function won't show up. I was thinking there'd be some value in seeing what assertion you're making in the last frame, but you wouldn't, and it will be in the next frame up anyway.

petroav commented:
lgtm

Commit:
Include pa log in test_status assertion failure messages.

test_status relies on comparing the actual output of presto-admin server status
to the expected output. Unfortunately, the status command is written in such a
way that some failures in PrestoClient are logged and otherwise silently
ignored. This results in incomplete output and an assertion failure with no
information about what the failure actually was. Since this is intermittent
and usually happens on Jenkins jobs, we don't get a chance to look at the logs
manually. The solution is to include them in the assertion failure message.
